nix-repl.el

```diff
@@ -34,10 +34,30 @@
     (define-key map "\t" 'completion-at-point)
     map))
 
+(defun nix-repl-save-all-histories ()
+  "Call `comint-write-input-ring' for all `nix-repl-mode' buffers."
+  (dolist (buf (buffer-list))
+    (with-current-buffer buf
+      (when (eq major-mode 'nix-repl-mode)
+        (comint-write-input-ring)))))
+
 (define-derived-mode nix-repl-mode comint-mode "Nix-REPL"
   "Interactive prompt for Nix."
   (setq-local comint-prompt-regexp nix-prompt-regexp)
   (setq-local comint-prompt-read-only t)
+  (let* ((is-remote (file-remote-p default-directory))
+         (maybe-xdg-data-home (shell-command-to-string "echo -n $XDG_DATA_HOME"))
+         (path-prefix (if (string-empty-p maybe-xdg-data-home)
+                          "~/.local/share"
+                        maybe-xdg-data-home))
+         (history-path (concat
+                        is-remote
+                        path-prefix
+                        "/nix/repl-history")))
+    (setq-local comint-input-ring-file-name history-path))
+  (comint-read-input-ring t)
+  (add-hook 'kill-buffer-hook #'comint-write-input-ring nil 'local)
```

Comment:
shell-mode does this like this:

```
(set-process-sentinel (get-buffer-process (current-buffer))
				#'shell-write-history-on-exit)
```

Do you think that might be a better hook? (replacing shell-write-history-on-exit with comint-write-input-ring that is) That would write out after `:q` or `Ctrl-D`.

Reply:
With what I submitted (which I took from [`racket-mode`'s REPL](https://github.com/greghendershott/racket-mode/blob/337d8f60c90a69b12c64c579c628f2388e16a7cb/racket-repl.el#L996)), I noticed that it was already writing out after `:q` and `Ctrl-D`. But I couldn't figure out how to write out after `C-x k` in the Nix REPL without modifying `kill-buffer-hook`, and using `(set-process-sentinel (get-buffer-process (current-buffer)) #'comint-write-input-ring)` in place of this does not appear to write after `C-x k`.